I have added filter options to see events virtual/in person

abhijain2003 opened the pull request:
<!-- If your PR fixes an open issue, use `Closes #999` to link your PR with the issue. #999 stands for the issue number you are fixing -->

## Fixes Issue

#2528 

## Changes proposed

<!-- List all the proposed changes in your PR -->
-> initialize state variable in events.js file with toggle values "virtual"/"person"
-> passing setState method in eventKey.js file as a props so I can toggle values on button clicks
-> implementing filter method before map in events.js file based on value of state variable
-> added a scale-105 class on hover in EventKey.js file

## Screenshots

https://user-images.githubusercontent.com/91718608/210301611-dd010e99-335f-480d-92a9-23702ee60537.mp4




<a href="https://gitpod.io/#https://github.com/EddieHubCommunity/LinkFree/pull/2605"><img src="https://gitpod.io/button/open-in-gitpod.svg"/></a>

eddiejaoude commented on components/event/EventKey.js:
```diff
@@ -1,10 +1,15 @@
-export default function EventKey() {
+export default function EventKey({ onChange }) {
+
+  function handleChangeType(value) {
```
I don't think this is needed, `onChange` can be called directly

eddiejaoude commented on components/event/EventKey.js:
```diff
@@ -1,10 +1,15 @@
 import { MdOutlineOnlinePrediction, MdOutlinePeople } from "react-icons/md";
 
-export default function EventKey() {
+export default function EventKey({ onChange }) {
```
I think `onChange` should be named more descriptively

eddiejaoude commented on components/user/UserEvents.js:
```diff
@@ -1,17 +1,26 @@
 import EventCard from "../event/EventCard";
 import Alert from "../Alert";
 import EventKey from "../event/EventKey";
+import { useState } from "react";
```
This doesn't seem to be used?

abhijain2003 replied:
I didn't get you can it be done without useState variables

eddiejaoude replied:
This import line is added to the file but I don't see this import used in the file?

eddiejaoude commented on components/user/UserEvents.js:
```diff
@@ -1,17 +1,26 @@
-          ))}
+          )
+            )
+          })}
```
These changes seem to be the same but with a console log?

eddiejaoude commented on pages/events.js:
```diff
@@ -20,6 +21,8 @@ export async function getServerSideProps(context) {
 }
 
 export default function Events({ events }) {
+  console.log('events', events)
```
Debug code should be removed

eddiejaoude commented on pages/events.js:
```diff
@@ -20,6 +21,8 @@ export async function getServerSideProps(context) {
 
 export default function Events({ events }) {
+  console.log('events', events)
+  const [eventType, seteventType] = useState('virtual');
```
By default it should not be filtered and should show all events

eddiejaoude commented on pages/events.js:
```diff
@@ -30,11 +33,19 @@ export default function Events({ events }) {
         <ul role="list" className="divide-y divide-gray-200">
-          {events.map((event, index) => (
+          {events
+          .filter((itm, _) => {
```
what is the `_` for?

abhijain2003 commented:
can you give me one more chance I'll resolve all conversation

eddiejaoude reviewed (changes requested):
Thank you for the contribution 👍 

- I think the `package-lock` file does not need to be changed
- I left some inline comments

eddiejaoude commented:
> can you give me one more chance I'll resolve all conversation

of course, we are here to help, any questions let us know 👍

abhijain2003 commented:
can I add a small button to take back to previous stage and show all events

abhijain2003 commented:
I have refactor the code
remove useless function in EventKey.js and calling onChange Directly
renamed onChange to onToggleEventType
renamed child inside filter method from itm to event
replaced _ with index
displaying all events by default 

didn't made any changes in package-lock previously so didn't toched it now as well
I don't think it can be done without useState variable so, I did not changed it

Can I commit it or would you like to give me more advices

eddiejaoude commented:
Great! Always commit otherwise we can't see the changes - GitHub will append them to this PR for review

ChinmayMhatre commented:
> I have refactor the code remove useless function in EventKey.js and calling onChange Directly renamed onChange to onToggleEventType renamed child inside filter method from itm to event replaced _ with index displaying all events by default
> 
> didn't made any changes in package-lock previously so didn't toched it now as well I don't think it can be done without useState variable so, I did not changed it
> 
> Can I commit it or would you like to give me more advices

Hey @abhijain2003 , I think what Eddie means. By the `useState`  is that the import isn't used in `UserEvent.js` so you could remove it. And Also that the `console.log` should be removed from everywhere after you are done debugging.

abhijain2003 commented:
ok
I just solved all issues hope this is ready to merge now

ChinmayMhatre commented:
I think we should add a border to the selected event type. What do you think @abhijain2003 , @eddiejaoude ?

ChinmayMhatre commented:
> ok I just solved all issues hope this is ready to merge now

Hey , I think eddie also had suggested to remove the package-lock.json from the pr.

abhijain2003 commented:
> I think we should add a border to the selected event type. What do you think @abhijain2003 , @eddiejaoude ?

I added a kind of scale property on hover on selected button

eddiejaoude commented:
> I think we should add a border to the selected event type

I think this can come in another PR

abhijain2003 commented:
so will you merge it now

eddiejaoude commented:
> so will you merge it now

@abhijain2003 please review remaining inline comments

abhijain2003 commented:
ok eddie

abhijain2003 commented:
eddie as you told me I have removed package-lock file and made a new commit 

hope there are no further checks are remaining

eddiejaoude commented:
> eddie as you told me I have removed package-lock file and made a new commit

Please do not remove this file, it is important, I asked to remove your changes to the file. You can use `git checkout main --- package-lock.json` and this should return the file to the state before your changes.

Also please address the remaining 2 inline comments that are not in a resolved state yet (these are the ones that are still expanded on this page (the others are collapsed as they are resolved)

abhijain2003 commented:
hey Eddie,  I think I messed up the contribution, but I resolved all conversations now, 

Thanks for Helping

eddiejaoude commented:
@abhijain2003 you marked this as resolved without a comment or changing the lines of code it is referring to, this makes it difficult to review. Do you need some help with this? I have marked it back to unresolved.

<img width="895" alt="Screenshot 2023-01-04 at 09 53 13" src="https://user-images.githubusercontent.com/624760/210529134-1487a337-8fb6-4fbe-a081-d6d082c9a984.png">

abhijain2003 commented:
not at all eddie this button was by mistaken pressed I am sorry for that,

and I made changes in code, removed unused imports and this issue is also resolved

abhijain2003 commented:
done eddie , i commited to my branch all changes

hope now it is resolved

eddiejaoude commented on pages/events.js:
```diff
@@ -20,6 +21,8 @@ export async function getServerSideProps(context) {
 
 export default function Events({ events }) {
+
+  const [eventType, seteventType] = useState('');
```
`useState` will need an import

abhijain2003 replied:
thank eddie, for letting me know I just added it

eddiejaoude commented on pages/events.js:
```diff
@@ -30,13 +33,23 @@ export default function Events({ events }) {
-            <EventCard event={event} username={event.username} key={index} />
-          ))}
+          {events
+            .filter((event, index) => {
```
I don't think `index` is used here?

abhijain2003 replied:
yes you are right, I have removed it

eddiejaoude commented:
Thank you 👍  I will resolve conflicts

abhijain2003 commented:
thanks to you, you really helped me in my initial stage of open source looking forward to contribute more

eddiejaoude commented:
Thank you for your hard work @abhijain2003 💪
